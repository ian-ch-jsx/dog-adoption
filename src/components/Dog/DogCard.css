.dog-card {
  margin: 20px;
  width: 300px;
  height: 350px;
  padding: 40px 40px 30px 40px;
  background-color: #d6ccc2;
  border: 1px solid #d4c5b8;
}

.dog-card:hover {
  background-color: #d5bdaf;
  filter: drop-shadow(4px 4px 3px #b6a391);
}

.image-preview {
  width: 250px;
  height: 250px;
  object-fit: cover;
  filter: sepia(0.5) grayscale(0.4) drop-shadow(2px 2px 3px #b6a391);
  border-radius: 0%;
}
