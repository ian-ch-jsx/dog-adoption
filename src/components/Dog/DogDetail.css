.dog-info {
  display: flex;
  flex-direction: column;
  margin: auto auto 40px auto;
  align-items: center;
  width: 700px;
  padding: 20px 50px 20px 50px;
  background-color: #d6ccc2;
}

.dog-image {
  margin: 10px;
  width: 500px;
  filter: drop-shadow(2px 2px 3px #b6a391);
}
