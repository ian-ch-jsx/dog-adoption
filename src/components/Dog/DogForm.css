body {
  color: #5c4c3d;
}

input {
  margin: 10px;
  width: 200px;
  border: 0px;
  background-color: #d6ccc2;
  color: #5c4c3d;
  padding: 10px 15px 10px 15px;
  border-radius: 50px;
}

.control {
  display: flex;
  flex-direction: column;
  align-items: center;
}
