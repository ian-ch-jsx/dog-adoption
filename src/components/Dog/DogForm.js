import React from 'react';

export default function DogForm({
  name,
  setName,
  bio,
  setBio,
  image,
  setImage,
  age,
  setAge,
  breed,
  setBreed,
  species,
  setSpecies,
  updateButton,
}) {
  return (
    <div className="update">
      <form>
        <div className="control">
          <label>name:</label>
          <input
            placeholder="Name"
            type="text"
            value={name}
            onChange={(e) => {
              setName(e.target.value);
            }}
          />
        </div>
        <div className="control">
          <label>species:</label>
          <input
            placeholder="species"
            type="text"
            value={species}
            onChange={(e) => {
              setSpecies(e.target.value);
            }}
          />
        </div>
        <div className="control">
          <label>breed:</label>
          <input
            placeholder="Name"
            type="text"
            value={breed}
            onChange={(e) => {
              setBreed(e.target.value);
            }}
          />
        </div>
        <div className="control">
          <label>age:</label>
          <input
            placeholder="age"
            type="number"
            value={age}
            onChange={(e) => {
              setAge(e.target.value);
            }}
          />
        </div>
        <div className="control">
          <label>image(url):</label>
          <input
            placeholder="image"
            type="text"
            value={image}
            onChange={(e) => {
              setImage(e.target.value);
            }}
          />
        </div>
        <div className="control">
          <label>bio:</label>
          <input
            placeholder="description"
            type="text"
            value={bio}
            onChange={(e) => {
              setBio(e.target.value);
            }}
          />
        </div>
        <div className="button">
          <button onClick={updateButton}>Save</button>
        </div>
      </form>
    </div>
  );
}
