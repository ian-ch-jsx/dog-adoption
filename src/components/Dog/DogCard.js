import React from 'react';
import './DogCard.css';
export default function DogCard({ name, image, breed, age }) {
  return (
    <div className="dog-card">
      <img src={image} className="image-preview"></img>
      <h1>{name}</h1>
      <p>
        {age} year old {breed}
      </p>
    </div>
  );
}
