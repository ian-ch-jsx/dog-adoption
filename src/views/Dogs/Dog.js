import React from 'react';
import DogCard from './DogCard';

export default function Dog() {
  return <div>help</div>;
}
