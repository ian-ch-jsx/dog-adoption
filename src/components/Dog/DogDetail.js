import React from 'react';
import './DogDetail.css';
export default function DogDetail({ name, image, breed, age, bio }) {
  return (
    <div className="dog-info">
      <h1>{name}</h1>
      <img src={image} className="dog-image"></img>
      <h2>{breed}</h2>
      <p>{age} year(s) old.</p>
      <p>{bio}</p>
    </div>
  );
}
