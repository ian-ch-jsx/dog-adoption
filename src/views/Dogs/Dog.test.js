import { render } from '@testing-library/react';
import Dog from './Dog';

test.skip('renders dog info', () => {
  const { container } = render(
    <Dog
      name="Barton"
      image="https://placedog.net/500?id=6"
      age={1}
      breed="Weimaraner"
      bio="dog is cute"
    />
  );
  expect(container).toMatchSnapshot();
});
