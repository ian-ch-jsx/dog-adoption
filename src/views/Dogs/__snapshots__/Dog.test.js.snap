// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dog info 1`] = `
<div>
  <div
    class="detail-container"
  >
    <div
      class="dog-info"
    >
      <h1>
        Lunchbox
      </h1>
      <img
        class="dog-image"
        src="https://placedog.net/1000/563?id=28"
      />
      <h2>
        Samoyed
      </h2>
      <p>
        0
         year(s) old.
      </p>
      <p>
        Lunchbox is an energetic, loving pup with no fear and a ton of fur.
      </p>
    </div>
    <a
      href="/dogs/edit/1"
    >
      <button>
        edit pet
      </button>
    </a>
    <button>
      Delete Pet
    </button>
  </div>
</div>
`;
